app/controllers/beers_controller.rb

```diff
@@ -0,0 +1,55 @@
+class BeersController < ApplicationController
+  def index
+    @beers = Beer.page(params[:page]).order(:name)
+  end
+
+  def show
+    @beer = Beer.find(params[:id])
+  end
+
+  def new
+    @beer = Beer.new
+  end
+
+  def create
+    @beer = Beer.new(beer_params)
```

Comment:
Where is the user authentication? How do you prevent unauthorized users from accessing this action?

```ruby
class BeersController < ApplicationController
  before_action :authenticate_user!, except: [:index, :show]

  def new
    @beer = Beer.new
  end

  def create
    @beer = Beer.new(beer_params)
    @beer.user = current_user

    if @beer.save
    # ...
```